item-api/src/main/java/com/example/auction/item/api/DeliveryOption.java

```diff
@@ -1,16 +1,16 @@
 package com.example.auction.item.api;
 
+import lombok.EqualsAndHashCode;
 import lombok.Value;
 import java.util.Optional;
 import java.util.UUID;
 
-public abstract class DeliveryOption {
+public interface DeliveryOption {
```

Comment:
Having these as abstract classes instead of interfaces was generating a Lombok warning:

```
Generating equals/hashCode implementation but without a call to superclass, even though this class does not extend java.lang.Object. If this is intentional, add '@EqualsAndHashCode(callSuper=false)' to your type.
```

Also, in some places, we were using interfaces and at others abstract classes. This makes all of them consistent.